lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -36,94 +137,180 @@ class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
+    // and kW, respectively, as described in the algorithm above.
+    SmallVector<AffineMap> indexingMaps = {
+        AffineMap::get(6, 0, inputExprs, op.getContext()),
+        AffineMap::get(6, 0, kernelExprs, op.getContext()),
+        AffineMap::get(6, 0, outputExprs, op.getContext())};
+    SmallVector<StringRef> iteratorTypes(6, getParallelIteratorTypeName());
+
+    // Input format is : [N, C, H, W]
+    Value inputShapeW = getDimOp(rewriter, loc, self, 3);
+
+    Value indicesResult =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, /*resultTensorTypes=*/indicesTensor.getType(),
+                /*inputs=*/ValueRange({maxPool2d, windowTensor}),
+                /*outputs=*/indicesTensor,
+                /*indexingMaps=*/indexingMaps,
+                /*iteratorTypes=*/iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value i = b.create<linalg::IndexOp>(loc, 0);
+                  Value j = b.create<linalg::IndexOp>(loc, 1);
+                  Value m = b.create<linalg::IndexOp>(loc, 2);
+                  Value n = b.create<linalg::IndexOp>(loc, 3);
+                  Value p = b.create<linalg::IndexOp>(loc, 4);
+                  Value r = b.create<linalg::IndexOp>(loc, 5);
+
+                  Value mTimesStride =
+                      b.create<arith::MulIOp>(loc, m, stride[0]);
+                  Value pTimesDilation =
+                      b.create<arith::MulIOp>(loc, p, dilation[0]);
+                  Value indexH = b.create<arith::AddIOp>(loc, mTimesStride,
+                                                         pTimesDilation);
+                  Value nTimesStride =
+                      b.create<arith::MulIOp>(loc, n, stride[1]);
+                  Value rTimesDilation =
+                      b.create<arith::MulIOp>(loc, r, dilation[1]);
+                  Value indexW = b.create<arith::AddIOp>(loc, nTimesStride,
+                                                         rTimesDilation);
+                  Value input = b.create<tensor::ExtractOp>(
+                      loc, paddedInput, ValueRange{i, j, indexH, indexW});
+                  Value maxVal = args[0];
+                  Value pred = b.create<arith::CmpFOp>(
+                      loc, arith::CmpFPredicate::OEQ, input, maxVal);
+
+                  Value indexHMinusPadding =
+                      b.create<arith::SubIOp>(loc, indexH, padding[0]);
+                  Value indexWMinusPadding =
+                      b.create<arith::SubIOp>(loc, indexW, padding[1]);
+                  Value outIndex = b.create<arith::MulIOp>(
+                      loc, indexHMinusPadding, inputShapeW);
+                  outIndex = b.create<arith::AddIOp>(loc, outIndex,
+                                                     indexWMinusPadding);
+
+                  Value result = b.create<arith::SelectOp>(
+                      loc, pred, castIndexToInt(b, loc, outIndex), args[2]);
```

Comment:
nit: assign args[2] to a variable with name to improve readability.  Can you put all the assignment at the start of the block? Similar to this code here https://github.com/llvm/torch-mlir/blob/5b9bdfaf3fb56c5aed12d689e21c9884b1c5a789/lib/Conversion/TorchToLinalg/Linear.cpp#L226.